### PR TITLE
Add OpenNebula one_image_facts module

### PR DESCRIPTION
**Pull request by ilicmilan**
##### SUMMARY
Add `one_image_facts` module for gathering facts about images on the [OpenNebula](https://opennebula.org/) open source cloud platform.

##### ISSUE TYPE
 - New Module Pull Request

##### COMPONENT NAME
* one_image_facts

##### ADDITIONAL INFORMATION
Add integration tests, as well

### CODE REVIEW

File: lib/ansible/modules/cloud/opennebula/one_image_facts.py [hunk 1]
```diff
@@ -0,0 +1,290 @@
...
+    description:
+      - Password of the user to login into OpenNebula RPC server. If not set
+      - then the value of the C(ONE_PASSWORD) environment variable is used.
+  ids:
+    description:
+      - A list of images ids whose facts you want to gather.
+  name:
+    description:
+      - A C(name) of the image whose facts will be gathered.
+      - If the C(name) begins with '~' the C(name) will be used as regex pattern
```
Comment (resmo) [inline lib/ansible/modules/cloud/opennebula/one_image_facts.py]: interesting approach.

File: lib/ansible/modules/cloud/opennebula/one_image_facts.py [hunk 2]
```diff
@@ -0,0 +1,290 @@
...
+
+    return auth_params(url=url, username=username, password=password)
+
+
+def main():
+    fields = {
+        "api_url": {"required": False, "type": "str"},
+        "api_username": {"required": False, "type": "str"},
+        "api_password": {"required": False, "type": "str", "no_log": True},
+        "ids": {"required": False, "type": "list"},
```
Comment (resmo) [inline lib/ansible/modules/cloud/opennebula/one_image_facts.py]: Just a small hint of "ansible magic": I would add an alisas e.g `aliases=['id']`. Ansible allows to use a single item for a list type: 

~~~yaml
- one_image_facts:
    ids: 124
~~~

with an alias `id` it makes perfect sense:

~~~yaml
- one_image_facts:
    id: 124
~~~

also update the docs accordenly:

~~~diff
  ids:
    description:
      - A list of images ids whose facts you want to gather
+   aliases: [ id ]
~~~

File: test/legacy/roles/one_image_facts/tasks/main.yml [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+
+- name: Verify that images list is empty
+  assert:
+    that:
+      - not images_with_regex is changed
+      - images_with_regex.images|length == 0
+
+- name: Try to fetch non-existent image by id
+  one_image_facts:
+    id: -999
```
Comment (resmo) [inline test/legacy/roles/one_image_facts/tasks/main.yml:64]: hmm, seems `id` is already used in the tests, but aliases is missing?

File: lib/ansible/modules/cloud/opennebula/one_image_facts.py (hunk 2 above)
Comment (ilicmilan) [reply lib/ansible/modules/cloud/opennebula/one_image_facts.py]: I have already perform the "ansible magic" for `one_vm` module.
It's perfectly suitable here.

File: test/legacy/roles/one_image_facts/tasks/main.yml (hunk 3 above)
Comment (ilicmilan) [reply test/legacy/roles/one_image_facts/tasks/main.yml:64]: There was an idea for an alias, but it's typo.

File: lib/ansible/modules/cloud/opennebula/one_image_facts.py [hunk 4]
```diff
@@ -0,0 +1,291 @@
...
+  register: result
+
+# Print all images facts
+- debug:
+    msg: result
+
+# Gather facts about an image using ID
+- one_image_facts:
+    ids:
+      - 123
```
Comment (resmo) [inline lib/ansible/modules/cloud/opennebula/one_image_facts.py:82]: probably using the alias here and without a list?

**Review by resmo — APPROVED** [review]
no reason to hold this back. 
shipit

**Comment by rvalle** [comment]
hi @ilicmilan, @resmo 

We just noticed that you started this OpenNebula module recently.

We have created an alternative Python bindings for OpenNebula, which uses the XSDs files to generate and maintain python stubs:

https://github.com/OpenNebula/addon-pyone

You can read about the module announcement here:

https://forum.opennebula.org/t/new-addon-pyone/5722

And status here:

https://forum.opennebula.org/t/pyone-progress-and-ansible-development-starting/5852

Were you aware of the pyone module?

We are now about to start pushing to the Ansible repository, and we noticed the potential name collision. Before using a different name to avoid it I wanted to reach out and see if you would be interested in joining forces, we are using the following repo for ansible:

https://github.com/privazio/ansible

And there is also a new inventory plugin on the works announced here:

https://forum.opennebula.org/t/opennebula-ansible-dynamic-inventory-script-feedback-needed/5788

Please let us know if you are interested in joining efforst and merging with us, or you prefer that we push to an alternative directory our opennebula modules.

**Pull request merged**